lib/ReactViews/Analytics/InvokeFunction.jsx

```diff
@@ -19,21 +19,26 @@ const InvokeFunction = React.createClass({
 
     componentWillMount() {
         this._parameterValues = {};
-        knockout.track(this, ['_parameterValues']);
-
-        this.initializeParameters(this.props);
+        this.defaultPropsIfNeeded(this.props);
+        this.enableContextItem(this.props);
     },
 
     componentWillUnmount() {
         this.removeContextItem();
     },
 
     componentWillReceiveProps(nextProps) {
-        this.initializeParameters(nextProps);
+        // This will be called when component is already mounted but props change, for example if you go from one WPS
+        // item to another. In that instance, we'll need to clear parameter values.
+        this._parameterValues = {};
+        this.defaultPropsIfNeeded(nextProps);
+        this.enableContextItem(nextProps);
     },
```

Comment:
By calling it in both places, it's behaving as it did before I made any changes. I'm not sure what it's doing though. Do you know?